service/client/client.js

```diff
@@ -128,8 +128,11 @@ export default class Client {
         }
 
         return {
-            headers,
-            ...options
+            ...options,
+            headers: {
```

Comment:
Can you move the part that changes the headers above like
```javascript
if (options.headers) {
    headers = Object.assign(headers, options.headers);
}
```
or something like that? Having it in the return statement makes it kind of difficult to follow.